ros/src/tl_detector/tl_detector.py

```diff
@@ -93,6 +94,13 @@ def image_cb(self, msg):
         :param msg: image from car-mounted camera
         :type msg: Image
         """
+        if self.last_state == TrafficLight.RED:
+            # if stopped, start driving only after detecting consistent non-red
+            STATE_COUNT_THRESHOLD = STATE_COUNT_THRESHOLD_DRIVE
+	else:	
+            # Stop on detecting a single instance of red
+            STATE_COUNT_THRESHOLD = STATE_COUNT_THRESHOLD_STOP
```

Comment:
generally variables in CAPS in python are constants, shouldnt be changed. 

Also, you are assuming when the last state is RED, the car is stopped. It could have detected the red lights from far away. Would that affect the program in any unforseen manner?

Reply:
No, it's good, even if the car sees red while not stopped, it will be slowing down, so it should not start accelerating again, until it sees 3 repeated instances of green, so the comment should be: "# if stopped or slowing down, start accelerating only after detecting consistent green (or consistent unknown)"

Reply:
Yeah you are right, but the point is, it's low only from far away, and the car needs to prepare and start braking early enough, that's why it should start braking at the slightest hint of red....but when you get near the light, the accuracy shoots up to 99%. But of course, if we have more data, it will detect even the far away lights with high confidence.

Reply:
- yeah I will change the all caps variable.

- No, it's good, even if the car is not stopped, it will be slowing down, so it should not start accelerating again until it sees 3 repeated instances of green, so the comment should be: "# if stopped or slowing down, start accelerating only after detecting consistent green (or consistent unknown)